@xen-orchestra/web-core/lib/components/ui/task-item/UiTaskItem.vue

```diff
@@ -0,0 +1,221 @@
+
+        <div class="content-right typo-body-regular-small">
+          <span v-if="task.end">
+            {{ `${t('task.ended')} ${end}` }}
+          </span>
+          <div class="progress">
+            <UiCircleProgressBar v-if="task.progress" :accent="progressAccent" size="small" :value="task.progress" />
+          </div>
+          <div class="actions">
+            <UiButtonIcon icon="fa:eye" size="medium" accent="brand" />
+            <div class="cancel">
+              <UiButtonIcon v-if="task.status === 'pending'" icon="fa:close" size="medium" accent="danger" />
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+    <template v-if="hasSubTasks && expanded">
+      <UiTaskList :tasks="subTasks" :depth="depth" />
+    </template>
+  </li>
+</template>
+
+<script lang="ts" setup>
+import UiCounter from '@core/components/ui/counter/UiCounter.vue'
+import UiLink from '@core/components/ui/link/UiLink.vue'
+import UiTaskList from '@core/components/ui/task-list/UiTaskList.vue'
+import { vTooltip } from '@core/directives/tooltip.directive'
+import { useTimeAgo } from '@vueuse/core'
+import { computed } from 'vue'
+import { useI18n } from 'vue-i18n'
+import UiButtonIcon from '../button-icon/UiButtonIcon.vue'
+import UiCircleProgressBar from '../circle-progress-bar/UiCircleProgressBar.vue'
+import UiInfo from '../info/UiInfo.vue'
+
+export type Task = {
+  id: string
+  infos?: { data: unknown; message: string }[]
+  name?: string
+  progress?: number
+  end?: number
+  status: 'failure' | 'interrupted' | 'pending' | 'success'
+  tasks?: Task[]
+  warning?: { data: unknown; message: string }[]
+}
+
+const { task } = defineProps<{
+  task: Task
+  expanded?: boolean
+  depth: number
+}>()
+
+const emit = defineEmits<{
+  expand: []
+}>()
+
+const { t } = useI18n()
+
+const subTasks = computed(() => task.tasks ?? [])
+
+const subTasksCount = computed(() => subTasks.value.length)
+
+const hasSubTasks = computed(() => subTasksCount.value > 0)
+
+const isError = computed(() => task.status === 'failure' || task.status === 'interrupted')
+
+const progressAccent = computed(() => (isError.value ? 'danger' : task.warning?.length ? 'warning' : 'info'))
+
+const end = useTimeAgo(() => task.end ?? 0)
+</script>
+
+<style lang="postcss" scoped>
+.ui-task-item {
+  &[data-depth='1']:last-child {
+    border-bottom: 0.1rem solid var(--color-neutral-border);
+  }
+
+  .container {
+    display: flex;
+    min-height: 4.8rem;
+    position: relative;
+
+    &::after {
+      content: '';
+      width: 100%;
+      position: absolute;
+      clip-path: inset(0 0 0 calc(4rem * v-bind(depth - 1)));
+      height: 0.1rem;
+      background: var(--color-neutral-border);
+    }
+
+    .tree-section {
+      display: flex;
+      align-items: center;
+      padding-left: 1.6rem;
+      gap: 0.6rem;
+
+      .tree-lines {
+        display: flex;
+        align-self: stretch;
+      }
+
+      .tree-line {
+        flex: 0 0 2.4rem;
```

Comment:
I think this value allows a better alignment of the vertical line:
```suggestion
        flex: 0 0 2.8rem;
```